athena-elasticsearch/src/main/java/com/amazonaws/connectors/athena/elasticsearch/AwsRestHighLevelClient.java

```diff
@@ -94,6 +102,33 @@ public AwsRestHighLevelClient(RestClientBuilder builder)
         return (LinkedHashMap<String, Object>) mappingsResponse.mappings().get(index).sourceAsMap();
     }
 
+    /**
+     * Retrieves cluster-health information for shards associated with the specified index. The request will time out
+     * if response is not received within 30 seconds.
+     * @param index is used to restrict the request to a specified index.
+     * @return a Map of cluster Ids with their associated health information.
+     * @throws IOException if an error occurs while sending the request to the Elasticsearch instance, the request
+     * times out, or no active-primary shards are present.
+     */
+    public Map<Integer, ClusterShardHealth> getShardHealthInfo(String index)
+            throws IOException
+    {
+        ClusterHealthRequest request = new ClusterHealthRequest(index).timeout(CLUSTER_HEALTH_TIMEOUT);
+        // Set request to shard-level details
+        request.level(ClusterHealthRequest.Level.SHARDS);
+
+        ClusterHealthResponse response = cluster().health(request, RequestOptions.DEFAULT);
+
+        if (response.isTimedOut()) {
+            throw new IOException("Request timed out.");
+        }
+        else if (response.getActivePrimaryShards() == 0) {
+            throw new IOException("There are no active primary shards.");
```

Comment:
I dont think this is an IOException, Runtime is probably more appropriate.

Reply:
Changed to RuntimeException.